src/main/java/com/cairn/waypoint/dashboard/endpoints/homeworkresponse/dto/AddHomeworkResponseEndpoint.java

```diff
@@ -0,0 +1,133 @@
+package com.cairn.waypoint.dashboard.endpoints.homeworkresponse.dto;
+import com.cairn.waypoint.dashboard.entity.HomeworkCategory;
+import com.cairn.waypoint.dashboard.entity.HomeworkQuestion;
+import com.cairn.waypoint.dashboard.endpoints.ErrorMessage;
+import com.cairn.waypoint.dashboard.entity.HomeworkResponse;
+import com.cairn.waypoint.dashboard.mapper.HomeworkResponseMapper;
+import com.cairn.waypoint.dashboard.repository.HomeworkCategoryRepository;
+import com.cairn.waypoint.dashboard.service.data.HomeworkQuestionDataService;
+import com.cairn.waypoint.dashboard.service.data.HomeworkResponseDataService;
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.media.Content;
+import io.swagger.v3.oas.annotations.media.Schema;
+import io.swagger.v3.oas.annotations.responses.ApiResponse;
+import io.swagger.v3.oas.annotations.security.SecurityRequirement;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import jakarta.transaction.Transactional;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.security.Principal;
+import java.time.LocalDateTime;
+import java.util.Optional;
+
+@Slf4j
+@RestController
+@Tag(name = "Homework Response")
+public class AddHomeworkResponseEndpoint {
+
+    public static final String PATH = "/api/homework-response";
+
+    private final HomeworkResponseDataService homeworkResponseDataService;
+    private final HomeworkQuestionDataService homeworkQuestionDataService;
+    private final HomeworkCategoryRepository homeworkCategoryRepository;
```

Comment:
Shouldn't be using repositories in your controller. Should be abstracted via the Service layer